envoy/auth_server/auth/endpoint_id_extractor.go

```diff
@@ -0,0 +1,81 @@
+package auth
+
+import (
+	"fmt"
+	"strings"
+
+	envoy_auth "github.com/envoyproxy/go-control-plane/envoy/service/auth/v3"
+)
+
+type EndpointIDExtractorType string
+
+const (
+	EndpointIDExtractorTypeURLPath EndpointIDExtractorType = "url_path"
+	EndpointIDExtractorTypeHeader  EndpointIDExtractorType = "header"
+)
+
+func (e EndpointIDExtractorType) IsValid() bool {
+	switch e {
+	case EndpointIDExtractorTypeURLPath, EndpointIDExtractorTypeHeader:
+		return true
+	default:
+		return false
+	}
+}
+
+// EndpointIDExtractor defines an interface for extracting an endpoint ID
+// from a given source, which could be a URL path or an HTTP header.
```

Comment:
```suggestion
// EndpointIDExtractor defines an interface for extracting an endpoint ID from a given source.
// This could be a URL path, HTTP header, etc...
```